sycl/doc/extensions/SubGroup/SYCL_INTEL_sub_group.asciidoc

```diff
@@ -85,13 +87,17 @@ class Functor
 
 It is illegal for a kernel or function to call a function with a mismatched sub-group size requirement, and the compiler should produce an error in this case.  The +reqd_sub_group_size+ attribute is not propagated from a device function to callers of the function, and must be specified explicitly when a kernel is defined.
 
+=== Compiler Flags
+
+The +-fsycl-core-sub-group-size+ flag compiles all kernels in the translation unit as though +[[intel::reqd_sub_group_size(core)]]+ was specified.
```

Comment:
What if we have a kernel which specified sub-group size?

For `auto` spec explicitly states above that `auto` can be overridden by a compiler flag. Overriding `core` with `core` shouldn't produce any problems, I guess, but what about particular specified sub-group sizes like `8`, for example?

Reply:
Good catch.  The intent here was to only override the behavior for kernels that do not already have an attribute specified.  This is why `auto` is specified both as the default behavior and an attribute: a user can write `[[reqd_sub_group_size(auto)]]` on a particular kernel to mean that they _always_ want the sub-group size to be selected by the implementation.

Is the change below sufficient to capture that?

```suggestion
The +-fsycl-core-sub-group-size+ flag compiles all kernels in the translation unit without a `reqd_sub_group_size` attribute as though +[[intel::reqd_sub_group_size(core)]]+ was specified.
```

Reply:
> Is the change below sufficient to capture that?

The idea LGTM, but in context of [this comment](https://github.com/intel/llvm/pull/2452#discussion_r487318374) from you it would probably better to refer to kernels with `auto` sub-group size instead of "without an attribute".

> I think this could be misleading. With the new wording, a requested size is always specified -- it's just that if no attribute is applied to the kernel, that size is `auto`.

Reply:
I like @Pennycook's proposed change, and I don't think it conflicts with the comment you reference.  Conceptually, the compiler first makes a pass over all kernels that have no explicit `[[intel::reqd_sub_group_size(core)]]` attribute, and sets the attribute to either `auto` or `core` depending on whether the `fsycl-core-sub-group-size` was specified.

Now, every kernel has a sub-group size defined and `info::kernel_device_specific::compile_sub_group_size` operates according to that size.

Reply:
Right.  I think the confusion here and the perceived incompatibility with https://github.com/intel/llvm/pull/2452#discussion_r487318374 arises from my lazy use of the word "specified".

If the user doesn't attach an attribute to a kernel, then either `auto` or `core` will be attached to it by default.  By the time the kernel has been compiled, the SPIR-V always has a sub-group size associated with it.

Adding a suggestion here and to the comment to try and avoid using the same word:
```suggestion
The +-fsycl-core-sub-group-size+ flag compiles all kernels in the translation unit without a `reqd_sub_group_size` attribute as though +[[intel::reqd_sub_group_size(core)]]+ was applied to the kernel.
```

Reply:
Tightened the description of the flag in e6dad9690e0836b228f7f158083f61566da5b0cd.